torch/_decomp/decompositions.py

```diff
@@ -1095,8 +1102,9 @@ def split(self: Tensor, split_size: int, dim: int = 0) -> List[Tensor]:
         assert dim_size == 0
         return [self]
     chunks = (dim_size + split_size - 1) // split_size
+    chunks = guard_int(chunks)
     split_sizes = [split_size for i in range(chunks)]
-    split_sizes[chunks - 1] = split_size - (split_size * chunks - dim_size)
+    split_sizes[-1] = split_size - (split_size * chunks - dim_size)
     return torch.split(self, split_sizes, dim)
```

Comment:
which tests do we currently run with decomps?

Reply:
split xfail was removed with this change, so this seems reasonable to me

Reply:
this is tested.